nilmtk/dataset_converters/dred/convert_dred.py

```diff
@@ -0,0 +1,137 @@
+'''
+DRED Dataset converter.
+The .h5 format is hosted in DRED official website. But the file is not fully compatible with NILMTK.
+Download the .h5 file directly or download .csv and convert to .h5 using this converter.
+
+https://drive.google.com/open?id=1NDiRGVb33SQaKL_W7Y6WXEs9_xR1xTij
+
+CSV file name :- csvmerged.csv
+.h5 file name :- book1DRED.h5
+
+'''
+
+from __future__ import print_function, division
+import pandas as pd
+import numpy as np
+from copy import deepcopy
+from os.path import join, isdir, isfile
+from os import listdir
+import fnmatch
+import re
+from sys import stdout
+from nilmtk.utils import get_datastore
+from nilmtk.datastore import Key
+from nilmtk.timeframe import TimeFrame
+from nilmtk.measurement import LEVEL_NAMES
+from nilmtk.utils import get_module_directory, check_directory_exists
+from nilm_metadata import convert_yaml_to_hdf5, save_yaml_to_datastore
+
+
+def convert_dred(input_path, output_filename, format='HDF'):
+    """
+    Parameters
+    ----------
+    input_path : str
+        The root path of the CSV files, e.g. House1.csv
+    output_filename : str
+        The destination filename (including path and suffix).
+    format : str
+        format of output. Either 'HDF' or 'CSV'. Defaults to 'HDF'
+    """
+        
+    # Open DataStore
+    store = get_datastore(output_filename, format, mode='w')
+
+    # Convert raw data to DataStore
+    _convert(input_path, store, 'Europe/Amsterdam')
+
+    # Add metadata
+    save_yaml_to_datastore(join(get_module_directory(), 
+                              'dataset_converters', 
+                              'dred', 
+                              'metadata'),
+                         store)
+    store.close()
+
+    print("Done converting DRED to HDF5!")
+
+def _convert(input_path, store, tz, sort_index=True):
+    """
+    Parameters
+    ----------
+    input_path : str
+        The root path of the DRED dataset.
+    store : DataStore
+        The NILMTK DataStore object.
+    measurement_mapping_func : function
+        Must take these parameters:
+            - house_id
+            - chan_id
+        Function should return a list of tuples e.g. [('power', 'apparent')]
+    tz : str 
+        Timezone e.g. 'Europe/Amsterdam'
+    sort_index : bool
+    """
+
+    check_directory_exists(input_path)
+
+    # Iterate though all houses and channels
+    # house 14 is missing!
+    houses = [1]
+    nilmtk_house_id = 0
+    for house_id in houses:
+        nilmtk_house_id += 1
+        print("Loading house", house_id, end="... ")
+        stdout.flush()
+        csv_filename = join(input_path, 'csvmerged' + '.csv')
```

Comment:
where is csvmerged.csv coming from. We should assume that the converter uses the dataset provided as is.

Reply:
The original .csv file did not have the UNIX timestamp column. It is being used by the converter to convert the time to a standard format. csvmerged.csv contains exactly the same data with an added 'unix' column.

Reply:
how do you get the "unix" column? Is it some transformation of data from the original .csv? Could you show the first 5 rows of original and csvmerged.csv?

Reply:
In this case, I think it will be better if the user gives the path of the file instead of providing the directory. As there is only one file.

Reply:
Original .csv :-

![image](https://user-images.githubusercontent.com/25877767/58757901-ae8eda00-8531-11e9-8399-32ca7ba998e1.png)

csvmerged .csv :-

![image](https://user-images.githubusercontent.com/25877767/58757912-e72eb380-8531-11e9-8481-cec5e10fda57.png)

The data is sampled at 1 sec. I converted the first time into its equivalent timestamp. Then used series to fill up the other rows.

Reply:
you do not need to convert to unix timestamp. Directly make use of the `pd.to_datetime` on the index column. Do ensure that the timezone is correctly interpreted and localized in the generated H5. So, the converter should take as input the DRED.csv provided by authors and output the H5.

Reply:
I tried to load using the string format timestamp. But, the dataframe will not be able to load at once. By using the unix timestamp, I reduced the size of the .csv file from 1.1 GB to around 750 Mb which is loaded with ease. The timezone is correctly interpreted and I have used Europe/Amsterdam as mentioned. Moreover, I tried to load in chunks and use 'append' instead of 'put' in HDFStore, but, it also shows memory overflow.

Reply:
You're on the right track with `append` instead of `put` and using load in chunks. Maybe, you need to finetune it to increase the number of chunks and ensure you do not get memory overflow.